### PR TITLE
added pagination and swagger

### PR DESCRIPTION
**Pull request by RedArtelerist**

### CODE REVIEW

**Review by mmosii — APPROVED** [review]
LGTM! Had no idea that swagger can be configed more)

File: src/main/java/maksym/fedorenko/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -20,28 +23,45 @@
...
     private final BookService bookService;
 
+    @Operation(
+            summary = "Retrieve all books",
+            description = "Receiving all books with pagination and sorting by fields."
+    )
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public List<BookDto> getAll(Pageable pageable) {
```
Comment (NazarSmith228) [inline src/main/java/maksym/fedorenko/bookstore/controller/BookController.java]: shouldn't you add `@PageableDefault` to your request parameter?

File: src/main/java/maksym/fedorenko/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -50,14 +70,25 @@ public BookDto updateBook(
...
     @ResponseStatus(HttpStatus.NO_CONTENT)
-    public void deleteBook(@PathVariable Long id) {
+    public void deleteBook(@PathVariable @Positive Long id) {
         bookService.delete(id);
     }
 
+    @Operation(
+            summary = "Search books by different parameters",
+            description = "Search books by title, specify a selection of authors, "
+                    + "and indicate the minimum and maximum price ranges."
```
Comment (NazarSmith228) [inline src/main/java/maksym/fedorenko/bookstore/controller/BookController.java]: minor

could you use multiline string here?

File: src/main/java/maksym/fedorenko/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -20,28 +23,45 @@
...
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
+    @Operation(
+            summary = "Retrieve a book by id",
+            description = "Get a book object by specifying its id."
+    )
     @GetMapping("/{id}")
     public BookDto getBookById(@PathVariable @Positive Long id) {
```
Comment (NazarSmith228) [inline src/main/java/maksym/fedorenko/bookstore/controller/BookController.java:50]: validation of @PathVariable,  @RequestParam etc. will not work untill you add @Validated annotation to your controller class. therefore this @Positive is not being checked


so pls add that annotation :)

**Review by NazarSmith228 — APPROVED** [review]
LGTM, but pls pay attention to my comments :)

**Pull request merged**